### ISSUE TITLE
NUnit3TestAdapter 4.0.0-beta.2 fails with Resharper

### ISSUE THREAD

**Issue by Sebazzz** [nunit/nunit3-vs-adapter#846]
## Issue
With NUnit3TestAdapter 4.0.0-beta.2, the Resharper test runner is not able to run any tests. It worked fine with 4.0.0-beta.1.

**v4.0.0-beta.1:**
![image](https://user-images.githubusercontent.com/1426097/115349475-30cf2880-a1b4-11eb-9aea-a8a48cfd0b31.png)

**v4.0.0-beta.2:**
![image](https://user-images.githubusercontent.com/1426097/115349943-bfdc4080-a1b4-11eb-993c-101cb02bf24e.png)

Both with **NUnit 3.13.1** on **.NET 5.0.5 (SDK 5.0.202)** with **Microsoft.NET.Test.Sdk 16.9.4**.

**Seems to happen on any project:**
[repro.zip](https://github.com/nunit/nunit3-vs-adapter/files/6341236/repro.zip)

## Environment
**Visual Studio:**

```
Microsoft Visual Studio Enterprise 2019
Version 16.9.3
VisualStudio.16.Release/16.9.3+31129.286
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Architecture Diagrams and Analysis Tools   00435-60000-00000-AA252
Microsoft Architecture Diagrams and Analysis Tools

Visual C++ 2019   00435-60000-00000-AA252
Microsoft Visual C++ 2019

.NET Portability Analyzer   1.1.10808.0
Evaluates portability of assemblies across .NET platforms.

ASP.NET and Web Tools 2019   16.9.693.2781
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2112521+5741df381174d72f08e3632bb99f52e8635b6a1a
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.9.693.2781
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.9.693.2781
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.9.693.2781
Azure Functions and Web Jobs Tools

C# Tools   3.9.0-6.21160.10+59eedc33d35754759994155ea2f4e1012a9951e3
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

CSS Tools 2019   1.1.4
Provides additional features to the CSS editor in Visual Studio.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

JetBrains ReSharper 2021.1.1   Build 211.0.20210409.154115
JetBrains ReSharper package for Microsoft Visual Studio. For more information about ReSharper, visit http://www.jetbrains.com/resharper. Copyright © 2021 JetBrains, Inc.

Markdown Editor   1.12.253
A full featured Markdown editor with live preview and syntax highlighting. Supports GitHub flavored Markdown.

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.40218.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NPM Task Runner   1.4.90
Adds support for npm scripts defined in package.json directly in Visual Studio's Task Runner Explorer. Includes full support for Yarn

NuGet Package Manager   5.9.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Project System Tools   1.0
Tools for working with C#, VisualBasic, and F# projects.

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Redgate SQL Change Automation   4.3.21108.25248
Extend DevOps processes to your SQL Server databases and safely automate database deployments.

Visit https://www.red-gate.com/sca/productpage for more information.

Copyright (C) Red Gate Software Ltd 2011-2021. All rights reserved.

This software contains components from Component Owl.
SQL Server is a registered trademark of Microsoft Corporation.
Visual Studio is a registered trademark of Microsoft Corporation.

SQL Change Automation contains code from the following open source software:

NuGet https://www.nuget.org/
SQL LocalDB Wrapper https://github.com/martincostello/sqllocaldb
Autofac https://autofac.org/
Json.NET https://json.net/
SemVer https://github.com/maxhauser/semver
Serilog https://serilog.net/
StringTemplate https://github.com/antlr/stringtemplate4
Extended WPF Toolkit https://wpftoolkit.codeplex.com/
Code InfoBox VSX http://www.codeproject.com/Articles/55196/Code-InfoBox-Visual-Studio-Extension-VSX
OctoPack https://github.com/OctopusDeploy/OctoPack
SQLite https://sqlite.org/

This product contains icons from http://www.visualpharm.com distributed under a free backlink license.

For license details or other notices relating to the above software, please see NOTICE.TXT and EULA.rtf in the SQL Change Automation application folder.
    

Redgate SQL Prompt   10.4.7.16599
Write, format, and refactor SQL effortlessly

RustLanguageExtensionOptionsPackage Extension   1.0
RustLanguageExtensionOptionsPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62103.10080
Microsoft SQL Server Data Tools

Syntax Visualizer   1.0
An extension for visualizing Roslyn SyntaxTrees.

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.30201.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.9.0-6.21160.10+59eedc33d35754759994155ea2f4e1012a9951e3
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.9.0-beta.21102.9+7ce7132f1459095e635194d09d6f73265352029a
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualizerPackage Extension   1.0
VisualizerPackage Visual Studio Extension Detailed Info

WebPack Task Runner   1.5.94
Adds support for WebPack in Visual Studio 2015 and 2017's Task Runner Explorer.

WiX Toolset Visual Studio Extension   1.0.0.4
WiX Toolset Visual Studio Extension version 1.0.0.4
Copyright (c) .NET Foundation and contributors. All rights reserved.

Yarn Installer   0.5.3
Makes it easy to restore npm packages using Yarn from within Visual Studio
```

**Resharper:**
```
JetBrains ReSharper 2021.1.1 Build 211.0.20210409.154115 built on 2021-04-09
dotCover 2021.1.20210409.162146
dotTrace 2021.1.20210409.162029
ReSharper C++ 2021.1.20210409.161944
ReSharper 2021.1.20210409.161805
dotMemory 2021.1.20210409.161945
```

**Comment by OsirisTerje** [comment]
Thanks!   This can be caused by the change to 3.12 NUnit engine.  And it should be raised as an issue with Jetbrains/Resharper.  NIce if you can do that, and link back here. :-)

**Comment by Sebazzz** [comment]
I didn't change the NUnit version, I only upgraded the adapter 😉

**Comment by OsirisTerje** [comment]
We reached out to Jetbrains and got the information this is fixed in version 2012.1.2, https://blog.jetbrains.com/dotnet/2021/04/23/resharper_rider_2021_1_2/  which was released April 22nd.  Go grab it :-)

**Comment by Sebazzz** [comment]
Thanks 😀
